src/content/post/20161122_drupal_application_bottlenecks_avoid_with_profiling.md

```diff
@@ -0,0 +1,152 @@
++++
+date = "2016-11-22T11:02:30Z"
+draft = false
+title = "Avoid bottlenecks on a Drupal web application and how identify it with profiling"
+description = "How identify bottleneck of your web application and avoid that can appear in the future"
+tags        = [ "profiling", "drupal", "performance", "web application"  ]
+topics      = [ "Profiling", "Drupal", "Performance" ]
+author = "Vinz"
++++
+
+### TL;DR
+
+This article is focused on the possible *bottlenecks of a web application*, how you can identify it to improve the performance of your code.
+
+We talk about profiling with *Blackfire*, PHP web app focused on *Drupal*. LAMP is the reference stack.
+
+<!-- more -->
+
+Application's performance is one of characteristics what separates a job from a good job, and demonstrates quality work.
+
+A customer's call about website performances, if server infrastructure is adequate to requests, most probably indicates some errors on our code.
+
+Google can be our friend if we find some 'web application performance checklist' but we have a problem, now. We need to identify and resolve our performance issues.
+
+A little disclaimer: this article is focused on a Drupal application, but information which follow are useful (I hope!) regardless of framework that you have used.
+
+### Application's check
+
+1. Check caches
+    1. Check Page cache, block cache and view cache
+2. Aggregate and minify CSS and js
+3. Deactivate development component
+    1. Modules like _devel_, _update_status_, _performance_ are really performances eaters.
+4. Deactivates _cron_ automation
+    1. Run tasks autonomously by night or in the hours with light traffic if it's possible. Remember to demand critical tasks or heavy tasks to external workers.
+5. Deactivates database logger
+    1. Move logs to _syslog_
+6. Check images size
+    1. Use opportune _image_styles_
+7. Check retrieval process of external resources (feed, streams, etc.)
+
+### Detection of application's bottleneck
+
+Once make previous checks, probably now our server take a sigh of relief, but _most_ probably this is not enough: we need with precision (specially if we don't have precise reference to slow pages) because we have an high server load. We need to collect data to use to apply right improvement, yes, I'm talking about [profiling](https://en.wikipedia.org/wiki/Profiling_(computer_programming)).
```

Comment:
Can't get what you mean with "We need with precision"...?
Moreover: "specially" (that means "appositamente") is not what you are searching for I guess... Especially is the right form to say "particolarmente".
To avoid redundancy I'd write `Especially if we lack specific references to slow pages`.